app/models/appointment.rb

```diff
@@ -103,22 +103,21 @@ def mark_patient_already_visited
   end
 
   def mark_patient_as_dead
-    self.patient.status = :dead
-    self.patient.save
+    patient.status = :dead
+    patient.save
```

Comment:
This type of change makes sense to me.  The `self` is redundant for method calls in ruby, _unless_ you operating only on `self`, for example:

```ruby
class Patient
  def assign=(val)
     @attributes[val] = val
  end

  def do_stuff
    self.assign = "value"
  end
```